lib/connections/service.go

```diff
@@ -324,6 +330,13 @@ func (s *service) handle(ctx context.Context) error {
 		rd, wr := s.limiter.getLimiters(remoteID, c, isLAN)
 
 		protoConn := protocol.NewConnection(remoteID, rd, wr, c, s.model, c, deviceCfg.Compression, s.cfg.FolderPasswords(remoteID))
+		go func() {
+			<-protoConn.Closed()
+			s.dialNowDevicesMut.Lock()
+			s.dialNowDevices[remoteID] = struct{}{}
+			s.scheduleDialNow()
+			s.dialNowDevicesMut.Unlock()
+		}()
```

Comment:
Could we get stuck in a fast infinite connection loop here if the remote side is dropping us due to paused or so?

Reply:
Yep, added a one minute minimum delay on closed connections. Pausing is done at hello stage, i.e. before this stage, but there still might be problems with cluster-config or a plainly misbehaving remote.